HRR16-TRex/TRex
# (refactor) Major raceController and server refactor

Server now handles races dynamically
- stores rooms, users and race data in a global var
- only emits data to clients that are a part of that room

Socket.io has been refactored into a factory on the racer controller

Test functionality:
1. open browser navigate to http://localhost:3030/#/raceView/test1
2. open another tab, navigate to http://localhost:3030/#/raceView/test1
3. open a 3rd tab, navigate to http://localhost:3030/#/raceView/test2

The first tab (first user to join a room is added as an admin) will be identified as the admin for the room. In the first tab, put in a time and click 'set time'. Go to tab #2 and confirm the countdown time is whatever was set on tab #1. If you check tab #3 you will see how the countdown time was NOT set. This is because tab#3 is a different room.

Go back to tab #1, click 'start clock'. Both tab #1 and tab #2 should be counting down and the race starts. Tab #3 should still be doing nothing.
